core/sql/cli/Globals.cpp

```diff
@@ -322,6 +310,18 @@ void CliGlobals::init( NABoolean espProcess,
     capacities_.setHeap(defaultContext_->exCollHeap());
     freespaces_.setHeap(defaultContext_->exCollHeap());
     largestFragments_.setHeap(defaultContext_->exCollHeap());
+    if (statsGlobals_ != NULL) 
+       memMonitor_ = statsGlobals_->getMemoryMonitor();
```

Comment:
Now that the memory monitor resides in another process, I think we need to make any of its methods that shouldn't be called by the executor thread private, like update() and updatePageFaultRate(). It would also be good to update the comments in memorymonitor.h to indicate which calls are meant for consumers and which ones are potentially in a separate thread.